cBioPortal/cbioportal-frontend
# Add transcript dropdown on result view

Fix: https://github.com/cBioPortal/cbioportal/issues/7479
Add transcript dropdown on result view, click dropdown to switch to other transcripts, the lollipop and mutation table will be updated based on the selection.

- Canonical transcript(default selection): the Protein Change and Mutation Type data are from database. The results view page should be exactly the same as the page without transcripts dropdown.
- Non-canonical transcripts: update mutations by genome nexus annotated results, so Protein Change and Mutation Type date are from Genome Nexus. Annotation column, Oncokb track and Cancer Hotspots tracks are disabled.

![image](https://user-images.githubusercontent.com/16869603/87186986-6f959100-c2ba-11ea-9954-9467bb5c6f00.png)

Mutation type column will show original data when it's `Other`
![image](https://user-images.githubusercontent.com/16869603/87984245-ac763a80-caa7-11ea-85a0-ca2bbe7aa6f9.png)


**How to test**:
https://deploy-preview-3211--cbioportalfrontend.netlify.app/
Try with any query, go to results view page, then go to Mutations tab. There should be a transcript switch dropdown on the right. (If no dropdown, run this in browser console)
```
localStorage.frontendConfig = '{"serverConfig":{"show_transcript_dropdown":true}}'
```
Switch transcripts by clicking the dropdown, after switching, check lollipop and mutation table, especially Protein Change, Mutation Type and HGVSc data.

It will be good to check Mutation Mapper Tool as well, same logic applied to it too.

**TODO**
- [x] Split out `Other` button to `Other` and `Fusion`
- [x] If the mutation is categorized as `Other`, show original data in Mutation Type column
- [x] Fixing protein change with `*` (like `* 280 *`) in server
